### ISSUE TITLE
SideNav fails to update active item when using BeforeLeaveEvent.postpone()

### ISSUE THREAD

**Issue by FelixGaebler** [vaadin/flow-components#5411]
### Description

When postponing navigation with a `BeforeLeaveListener` the `SideNavItem` is not shown as active even after proceeding.

### Expected outcome

The `SideNavItem` you selected should be active now.

### Minimal reproducible example

https://github.com/FelixGaebler/vaadin-bugs/tree/master/src/main/java/de/dataport/bugs/views/report4
```java
@Override
public void beforeLeave(BeforeLeaveEvent event) {
	BeforeLeaveEvent.ContinueNavigationAction action = event.postpone();

	ConfirmDialog dialog = new ConfirmDialog();
	dialog.setText("Navigate?");
	dialog.setConfirmButton("Yes", e -> action.proceed());
	dialog.setCancelText("No");
	dialog.setCancelable(true);
	dialog.open();
}
```

### Steps to reproduce

1. Create a new vaadin start project with 2-3 empty views.
2. Take one empty view, add `implements BeforeLeaveObserver` and add the minimal reproducible example code to the view.
3. Navigate to the customized view.
4. Navigate away from the customized view.
5. Click yes when the dialog shows up.

### Environment

Vaadin version(s): 24.1.5
OS: Ubuntu 22

**Comment by web-padawan** [comment]
Tested this with the latest alpha at the time of writing (24.2.0.alpha9) and now it works as expected:

https://github.com/vaadin/flow-components/assets/10589913/28f871c7-ced7-4faa-ace7-f6210930a4e4

**Comment by yuriy-fix** [comment]
Closing as fixed in 24.2.
In 24.1 SideNav was shipped as a preview feature, so the fix won't be backported there.
